debug/testlib.py

```diff
@@ -158,6 +157,31 @@ def __init__(self, cmd=None, config=None, debug=False):
                 messaged = True
                 print "Waiting for OpenOCD to examine RISCV core..."
 
+        self.port = self._get_gdb_server_port()
+
+    def _get_gdb_server_port(self):
+        """Get port that OpenOCD's gdb server is listening on."""
+        MAX_ATTEMPTS = 5
+        PORT_REGEX = re.compile(r'(?P<port>\d+) \(LISTEN\)')
+        for _ in range(MAX_ATTEMPTS):
+            with open(os.devnull, 'w') as devnull:
+                output = subprocess.check_output([
+                    'lsof',
+                    '-a',  # Take the AND of the following selectors
+                    '-p{}'.format(self.process.pid),  # Filter on PID
+                    '-iTCP',  # Filter only TCP sockets
+                ], stderr=devnull)
+            matches = list(PORT_REGEX.finditer(output))
```

Comment:
They're actually slightly different. [`regex.findall()`](https://docs.python.org/2/library/re.html#re.findall) returns a list of strings or a list of tuples of strings if you have multiple capture groups. [`regex.finditer()`](https://docs.python.org/2/library/re.html#re.finditer) yields `MatchObjects`, which allow you to use named capture groups like I did.